test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -1888,6 +1888,37 @@ def forward(self, l_y_: "f32[16, 16]"):
 """,
             )
 
+    def test_return_size(self):
+        def run(dynamic):
+            torch.compiler.reset()
+
+            @nested_compile_region
+            def gn(x):
+                y = x + 1
+                z = x.shape
+                return y, z
+
+            def fn(x):
+                z0 = gn(x)
+                z1 = gn(x)
+                return z0[0] + z1[0], z0[1]
+
+            x = torch.randn(8, 8, requires_grad=True)
+            x_clone = x.detach().clone().requires_grad_(True)
+            ref = fn(x)
+            opt_fn = torch.compile(
+                fn, backend="inductor", fullgraph=True, dynamic=dynamic
+            )
+            res = opt_fn(x_clone)
+            self.assertEqual(ref, res)
+
+            ref[0].sum().backward()
+            res[0].sum().backward()
+            self.assertEqual(x.grad, x_clone.grad)
+
```

Comment:
1) symints should be allowed to be outputs of the graph
2) any non-symbolic expressions (constants) should not be outputs of the graph. Instead, Dynamo should figure out how to propagate them along.

For example, if there's a function like the following:
```
def f(x):
    return x.sin(), 1
```
then the subgraph should just look like the following:
```
def subgraph(x):
    return torch.ops.aten.sin.default(x)
```
This is to avoid giving Inductor anything that looks crazy (a constant that it can't handle)